pkg/abstractions/image/build.go

```diff
@@ -321,6 +324,32 @@ func (b *Builder) genContainerId() string {
 	return fmt.Sprintf("%s%s", types.BuildContainerPrefix, uuid.New().String()[:8])
 }
 
+func (b *Builder) checkForInitialImagePullError(containerId string, outputChan chan common.OutputMsg) {
+	// If the initial image pull of custom base image fails, the container will never receive
+	// a container address for the runc client to connect to.
+	// If it fails without a container address, we can still look up the exit code
+	// to terminate the build and notify the user.
+	for {
```

Comment:
We might want to pass the calling function's context and check to see if its done using a select{} in the for loop. Doing this should guarantee that if someone disconnects or hits Ctl-C, this will exit regardless whats happening in the build workers.